core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -285,12 +285,24 @@ public CameraSource stopAsync() {
 
   @Override
   public void stopAndAwait() {
-    stopAsync().cameraService.stopAndAwait();
+    try {
+      stopAsync().cameraService.stopAndAwait();
+    } catch (IllegalStateException e) {
+      if (!cameraService.state().equals(State.FAILED)) {
+        throw e;
+      }
+    }
   }
 
   @Override
   public void stopAndAwait(long timeout, TimeUnit unit) throws TimeoutException {
-    stopAsync().cameraService.stopAndAwait(timeout, unit);
+    try {
+      stopAsync().cameraService.stopAndAwait(timeout, unit);
+    } catch (IllegalStateException e) {
+      if (!cameraService.state().equals(State.FAILED)) {
+        throw e;
+      }
```

Comment:
It means that the service crashed while running, or an exception was thrown on startup (e.g. the webcam doesn't exist or isn't available). All this does is stop if it's in terminal state (TERMINATED or FAILED), instead of throwing another exception if startup failed